lib/cache/ExpirationStrategy.ts

```diff
@@ -0,0 +1,40 @@
+import { StorageType } from './StorageType';
+import { MemoryStorage } from './MemoryStorage';
+
+export class ExpirationStrategy<K> {
```

Comment:
[SRP](https://github.com/labs42io/clean-code-typescript#single-responsibility-principle-srp) violated.
You should have two implementations, one for `SlidingExpiration` and another for `AbsoluteExpiration`.
Now you have a single class doing both things.